src/session/content/item_row.rs

```diff
@@ -94,16 +96,311 @@ impl ItemRow {
                 match item.type_() {
                     ItemType::Message(message) => {
                         let content = message.content().0;
-                        let message = message.to_owned().upcast();
+                        let sender_name = match message.sender().to_owned() {
+                            MessageSender::User(data) => super::item_row::sender_name(&data),
+                            MessageSender::Chat(data) => data.title(),
+                        };
+                        let message_obj = message.to_owned().upcast();
 
                         match content {
                             MessageContent::MessagePhoto(_) => {
-                                self.set_child_row::<MessagePhoto>(message)
+                                self.set_child_row::<MessagePhoto>(message_obj)
                             }
                             MessageContent::MessageSticker(data) if !data.sticker.is_animated => {
-                                self.set_child_row::<MessageSticker>(message)
+                                self.set_child_row::<MessageSticker>(message_obj)
+                            }
+                            MessageContent::MessageChatChangeTitle(data) => {
+                                let title = data.title;
+                                let child = self.get_or_create_event_row();
+                                let type_ = message.chat().type_().clone();
+                                match type_ {
+                                    ChatType::Supergroup(data) => {
+                                        let is_channel = data.is_channel();
+                                        if is_channel {
+                                            let msg =
+                                                gettext!("Channel name was changed to «{}»", title);
+                                            child.set_label(&format!("<b>{}</b>", msg))
+                                        } else if message.is_outgoing() {
+                                            let msg =
+                                                gettext!("You changed group name to «{}»", title);
+                                            child.set_label(&format!("<b>{}</b>", msg))
+                                        } else {
+                                            let msg = gettext!(
+                                                "{} changed group name to «{}»",
+                                                sender_name,
+                                                title
+                                            );
+                                            child.set_label(&format!("<b>{}</b>", msg))
+                                        }
```

Comment:
The code is duplicated. I've seen this pattern a lot of times in this PR.  Somewhere below, I gave an example how to use a `macch guard` to get rid of the duplication.